rust/kernel/error.rs

```diff
@@ -352,13 +352,22 @@ impl Error {
     pub fn to_kernel_errno(self) -> c_types::c_int {
         self.0
     }
+
+    #[cfg(not(testlib))]
+    fn errname(&self) -> *const c_types::c_char {
+        // SAFETY: Just an FFI call, there are no extra safety requirements.
+        unsafe { bindings::errname(-self.0) }
```

Comment:
Does `errname` allow passing in invalid error codes?

Reply:
Yes, it does, it returns `NULL` in such cases.

You can see the code [here](https://elixir.bootlin.com/linux/latest/source/lib/errname.c#L217).

Reply:
We might just want to make the signature `fn errname(&self) -> Option<&CStr>` or  `fn errname(&self) -> Option<&str>`.

Reply:
I mean it might be useful to expose `errname` to users anyway. Converting a pointer to `Option` is bad either.

Reply:
Oh, I see.

Yes, I think it would be helpful. I'll make this change.